client/my-sites/themes/themes-search-card/index.jsx

```diff
@@ -154,4 +160,20 @@ const ThemesSearchCard = React.createClass( {
 	}
 } );
 
-export default localize( ThemesSearchCard );
+export default connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+		const isJetpack = isJetpackSite( state, siteId );
+
+		let externalUrl;
+		if ( ! siteId ) {
+			externalUrl = oldShowcaseUrl;
+		} else if ( isJetpack ) {
+			externalUrl = getSiteAdminUrl( state, siteId, 'theme-install.php' );
```

Comment:
The _More_ button actually doesn't show for Jetpack sites.

<img width="840" alt="screen shot 2017-03-15 at 11 27 53" src="https://cloud.githubusercontent.com/assets/7767559/23946523/9877d02e-0972-11e7-9d33-f97e1eafe159.png">